wordpress-mobile/WordPress-iOS
# Jetpack Focus: Add analytics tracking for the Jetpack fullscreen overlay

Part of #19447

## Description
This PR tracks the fullscreen overlay events. 

## Testing Instructions

P.S: To be able to test, you can disable the frequency logic by changing the value of `secondsInDay` found in `JetpackOverlayFrequencyTracker.swift:91` to `0`.

1. Open the app
2. Make sure the "Jetpack Features Removal Phase one" flag is enabled from the debug menu
3. Navigate to stats
4. The overlay is displayed
5. Check logs for the event `remove_feature_overlay_displayed`
6. Make sure the event properties include phase and source and are of correct values
7. Tap the CTA button
8. Check logs for the event `remove_feature_overlay_button_tapped`
9. Make sure the event properties include phase and source and are of correct values
10. Tap the continue button
11. Check logs for the event `remove_feature_overlay_dismissed`
12. Make sure the event properties include phase, source and dismissal type and are of correct values
13. Display the overaly again
14. Tap the close button
15. Check logs for the event `remove_feature_overlay_dismissed`
16. Make sure the event properties include phase, source and dismissal type and are of correct values
17. Repeat the same for Reader and Notifications overlays

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
